File: notifyr/AppSettings.m
//
//  AppSettings.m
//  notifyr
//
//  Created by Nelson Narciso on 2015-02-21.
//  Copyright (c) 2015 Nelson Narciso. All rights reserved.
//

#import "AppSettings.h"

@implementation AppSettings

@synthesize deviceToken = _deviceToken, sentDeviceTokenToApi = _sentDeviceTokenToApi;

NSString * const kUserDefaultsKeyDeviceToken = @"deviceToken";
NSString * const kUserDefaultsKeySentDeviceTokenToApi = @"sentDeviceTokenToApi";


static AppSettings *sharedInstance = nil;

+ (AppSettings *)sharedSettings {
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        sharedInstance = [[AppSettings alloc] init];
    });
    return sharedInstance;
}


#pragma mark - deviceToken

- (NSString *)deviceToken
{
    if (_deviceToken == nil)
    {
        _deviceToken = [[NSUserDefaults standardUserDefaults] stringForKey:kUserDefaultsKeyDeviceToken];
    }
    return _deviceToken;
}

- (void)setDeviceToken:(NSString *)deviceToken
{
    if ([_deviceToken isEqualToString: deviceToken])
    {
        return;
    }
    _deviceToken = deviceToken;
    NSUserDefaults *userDefaults = [NSUserDefaults standardUserDefaults];
    [userDefaults setObject:deviceToken forKey:kUserDefaultsKeyDeviceToken];
    [userDefaults synchronize];
}


#pragma mark - sentDeviceTokenToApi

- (BOOL)sentDeviceTokenToApi
{
    _sentDeviceTokenToApi = [[NSUserDefaults standardUserDefaults] boolForKey:kUserDefaultsKeySentDeviceTokenToApi];
    return _sentDeviceTokenToApi;
}

- (void)setSentDeviceTokenToApi:(BOOL)sentDeviceTokenToApi
{
    if (_sentDeviceTokenToApi == sentDeviceTokenToApi)
    {
        return;
    }
    _sentDeviceTokenToApi = sentDeviceTokenToApi;
    NSUserDefaults *userDefaults = [NSUserDefaults standardUserDefaults];
    [userDefaults setBool:sentDeviceTokenToApi forKey:kUserDefaultsKeySentDeviceTokenToApi];
    [userDefaults synchronize];
}


@end

File: notifyr/AppSettings.swift
//
//  AppSettings.swift
//  notifyr
//
//  Created by Klein on 2015-02-21.
//  Copyright (c) 2015 Nelson Narciso. All rights reserved.
//

import UIKit

class AppSettings: NSObject {
    
    private let kUserDefaultsKeyDeviceToken = "deviceToken";
    private let kUserDefaultsKeySentDeviceTokenToApi = "sentDeviceTokenToApi"
    
    private var _deviceToken: String?
    
    var deviceToken: String? {
        
        get {
            if (_deviceToken == nil)
            {
                _deviceToken = NSUserDefaults.standardUserDefaults().stringForKey(kUserDefaultsKeyDeviceToken)
            }
            
            return _deviceToken;
        }
        
        set {
            if (_deviceToken == newValue) {
                return;
            }
            _deviceToken = newValue
            var userDefaults = NSUserDefaults.standardUserDefaults()
            userDefaults .setObject(_deviceToken, forKey: kUserDefaultsKeySentDeviceTokenToApi)
            userDefaults.synchronize()
        }
        
    }
    
    
    private var _sentDeviceTokenToApi: Bool?
    
    var sentDeviceTokenToApi: Bool {
        
        get {
            if (_sentDeviceTokenToApi == nil)
            {
                _sentDeviceTokenToApi = NSUserDefaults.standardUserDefaults().boolForKey(kUserDefaultsKeySentDeviceTokenToApi)
            }
            
            return _sentDeviceTokenToApi!;
        }
        
        set {
            if (_sentDeviceTokenToApi == newValue) {
                return;
            }
            _sentDeviceTokenToApi = newValue
            var userDefaults = NSUserDefaults.standardUserDefaults()
            userDefaults .setObject(_sentDeviceTokenToApi, forKey: kUserDefaultsKeySentDeviceTokenToApi)
            userDefaults.synchronize()
        }
        
    }
    
    
    class var sharedSettings : AppSettings {
        struct Static {
            static var onceToken : dispatch_once_t = 0
            static var instance : AppSettings? = nil
        }
        dispatch_once(&Static.onceToken) {
            Static.instance = AppSettings()
        }
        return Static.instance!
    }
    

}
